packages/data-stores/src/auth/actions.ts

```diff
@@ -77,24 +84,32 @@ const fetchWpLogin = ( action: WpLoginAction, params: object ): FetchWpLoginActi
 	} as const );
 
 export function* submitPassword( password: string ) {
+	yield clearErrors();
 	const username = yield { type: 'SELECT_USERNAME_OR_EMAIL' };
 
-	try {
-		const loginResponse = yield fetchWpLogin( 'login-endpoint', { username, password } );
+	const loginResponse = yield fetchWpLogin( 'login-endpoint', { username, password } );
```

Comment:
I'm kind of confused about this change, it looks like fetchWpLogin does not throw JS Errors if there is an unsuccessful request, I wonder if the original try/catch code would have worked at all? the code in the other stores still uses the try/catch pattern, do you think they should they be updated to use this pattern?

Reply:
The try/catch might still be useful for catching 500s / server timeouts. Would it work to move the `if loginResponse.ok` check to inside the `try` block as an additional `yield receiveWpLoginFailed` (or throw an exception and pass the `body` to the `catch` block?). Or is that overly complicating things?

Reply:
> I wonder if the original try/catch code would have worked at all?

Yeah the original `try/catch` wasn't really doing anything, I'd just copied it from the `submitUsernameOrEmail` code. The other stores use `try/catch` because their network requests use `wpcom-proxy-request` under the hood, which deals with non-200 responses by throwing.

The `wp-login.php` calls use `fetch()` so it's not throwing for 400s (which imo is the correct thing to do).

I think Andrew's right though, I should probably leave a `try/catch` block in place to translate any exceptions into Redux actions. I think the most likely exception would be calling `.json()` failing.